src/schema/v2/updateQuizMutation.ts

```diff
@@ -0,0 +1,39 @@
+import { GraphQLBoolean, GraphQLNonNull, GraphQLString } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { quiz, QuizType } from "./quiz"
+import { ResolverContext } from "types/graphql"
+
+export const updateQuizMutation = mutationWithClientMutationId<
+  {
+    artworkId: string
+    clearInteraction?: boolean
+    userId: string
+  },
+  any,
+  ResolverContext
+>({
+  name: "updateQuizMutation",
+  description: "Update a quiz artwork interacted_with flag",
+  inputFields: {
+    artworkId: { type: new GraphQLNonNull(GraphQLString) },
+    clearInteraction: { type: GraphQLBoolean },
+    userId: { type: new GraphQLNonNull(GraphQLString) },
```

Comment:
Do we actually need the `userId`? Braze doesn't need to execute these mutations, only a logged in user will be capable of doing it, right?

Reply:
Like theoretically, if this was like this I'd be able to login and update any user's quiz?

Reply:
I guess Gravity would handle the access control, of course. But it still doesn't seem necessary.

Reply:
https://github.com/artsy/gravity/blob/e0dae47a5028705a1eb33475c344f957752590c9/app/api/v1/user_art_quiz_endpoint.rb#L19

Reply:
This is going to be accessible to trusted client applications as well so that we can use it in email campaigns.

Reply:
I'm not so sure that's true — it's a mutation. Nothing in Braze is going to be mutating and the Gravity endpoint doesn't support arbitrary IDs unless I'm missing another endpoint?

Reply:
Can you point to the Gravity endpoint that you're hitting. The one I linked doesn't require it.

Reply:
And you don't need to pull it from anywhere. Gravity has the logged in user. That endpoint utilizes `current_user`